packages/mui-material/src/Stack/Stack.js

```diff
@@ -79,6 +79,17 @@ export const style = ({ ownerState, theme }) => {
       base,
     });
 
+    if (typeof directionValues === 'object') {
+      Object.keys(directionValues).forEach((breakpoint, index, breakpoints) => {
```

Comment:
Shouldn't we make sure first that these are sorted?

Reply:
`directionValues` is already sorted (by utility function [resolveBreakpointValues](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Stack/Stack.js#L72), so I think we are good!?

Reply:
Perfect, thanks for the explanation :)